### ISSUE TITLE
Type-only/Emit-less TS files aren't being checked

### ISSUE THREAD

**Issue by aomader** [jaredpalmer/tsdx#725]
### Current Behavior

After creating a simple `react-with-storybook` tsdx project (using `create`), add the following files:

**src/interfaces/test.ts**:
```ts
export interface Foo {
  bar: asd; // should not compile!
}
```

**src/interfaces/index.ts**:
```ts
export * from './test';
```

**src/index.tsx**:
```ts
// ...
export * from './interfaces';
```

Using this setup, running `npm run start` "successfully" builds the package, although it should error out due to the above error.

In contrast, when I run `tsc --noEmit`, the error is successfully recognized.

### Expected behavior

When running `npm run start`, I expect it to catch the error.

### Your environment

| Software         | Version(s) |
| ---------------- | ---------- |
| TSDX           | 0.13.2
| TypeScript       | 3.9.3
| npm        | 6.14.4
| Node             |  14.1.10
| Operating System | Win10

Find the minimal example from above here: 
[lib2.zip](https://github.com/jaredpalmer/tsdx/files/4657493/lib2.zip)

**Comment by agilgur5** [comment]
Sounds like this might be similar to #527 which never provided a repro. `test.ts` is a type-only import and its type isn't used anywhere, so it might just be pruned out by `rollup-plugin-typescript2` or `rollup` itself

**Comment by ezolenko** [comment]
Yeah, rollup prunes unused sources, so rtp2 plugin might never see something not imported by entry file.

**Comment by agilgur5** [comment]
I did some research on this in the past few weeks/months and found a reference to the original issue in `rollup-plugin-typescript` (https://github.com/rollup/rollup-plugin-typescript/issues/28) from [`@wessberg/rollup-plugin-ts`'s docs](https://github.com/wessberg/rollup-plugin-ts/blame/980e1dad284a7fc3cd3c2ec6c862be13099843ad/README.md#L42). That was then moved to https://github.com/rollup/plugins/issues/71 which was supposedly fixed in `@rollup/plugin-typescript` v3.0.0

We don't use `@rollup/plugin-typescript`, but that issue has references from other plugins, e.g. this one https://github.com/wessberg/rollup-plugin-ts/issues/29 which [has a comment](https://github.com/wessberg/rollup-plugin-ts/issues/29#issuecomment-536212388) that refers to `rollup-plugin-typescript2`'s issue: https://github.com/ezolenko/rollup-plugin-typescript2/issues/24 (which also referenced from the original issue).

@ezolenko thanks for responding here. Per that issue, I thought this was fixed by https://github.com/ezolenko/rollup-plugin-typescript2/commit/448913d411aa921c407240e517c43f041d65859b with [the current code here](https://github.com/ezolenko/rollup-plugin-typescript2/blob/2f51110a8cad186dffed8aea2834bd82c44209e5/src/index.ts#L331-L344)? Does that only handle entry files? Can we workaround that? It seems like `@wessberg/rollup-plugin-ts` has fixed the emit-less/type-only issue in somehow.

**Comment by agilgur5** [comment]
@ezolenko following up about the above

**Comment by agilgur5** [comment]
Noting here that this type-only issue was reproduced upstream in https://github.com/ezolenko/rollup-plugin-typescript2/issues/298.
I became a maintainer of rpt2 a couple of months ago and fixed that in https://github.com/ezolenko/rollup-plugin-typescript2/pull/345 , which was released in rpt2 [`0.33.0`](https://github.com/ezolenko/rollup-plugin-typescript2/releases/0.33.0.1)

> Not sure if [ezolenko/rollup-plugin-typescript2#211](https://github.com/ezolenko/rollup-plugin-typescript2/issues/211) is related given mention of "entry file"

This was actually the root cause and all of my references were related.
https://github.com/ezolenko/rollup-plugin-typescript2/pull/345 allowed a specific workaround to function for many use-cases, but I later fixed the root cause issue in https://github.com/ezolenko/rollup-plugin-typescript2/pull/406, which was released in rpt2 [`0.34.0`](https://github.com/ezolenko/rollup-plugin-typescript2/releases/tag/0.34.0).
